packages/auto-label/src/auto-label.ts

```diff
@@ -247,94 +221,78 @@ handler.addLabeltoRepoAndIssue = async function addLabeltoRepoAndIssue(
-      console.log(
-        `Auto-detected label ${autoDetectedLabel} for ${owner}/${repo}#${issueNumber}`
-      );
-      autoDetectedLabelAdded = await handler.maybeAddAPILabel(
+    autoDetectedLabel = handler.autoDetectLabel(jsonArray, issueTitle);
+  }
+  const colorNumber =
+    jsonArray?.findIndex((object: JSONData) => objectInJsonArray === object) %
+      colorsData.length || 0;
+
+  const githubLabel = objectInJsonArray?.github_label || autoDetectedLabel;
+
+  let alreadyExists;
+  if (githubLabel) {
+    alreadyExists = await handler.checkExistingLabels(
+      github,
+      owner,
+      repo,
+      githubLabel
+    );
+    if (alreadyExists === null || alreadyExists === undefined) {
+      handler.createLabel(
         github,
         owner,
         repo,
-        issueNumber,
-        labelsOnIssue,
-        autoDetectedLabel
+        githubLabel,
+        colorsData[colorNumber].color
       );
     } else {
       console.log(
-        `Auto-detect did not add a label for ${owner}/${repo}#${issueNumber}: '${issueTitle}'`
+        'This label already exists on the repository, will check if it also exists on the issue'
       );
     }
-    return;
-  }
-
-  const colorNumber = jsonArray.findIndex(
-    (object: JSONData) => objectInJsonArray === object
-  );
-  const githubLabel = objectInJsonArray.github_label;
-
-  const alreadyExists = await handler.checkExistingLabels(
-    github,
-    owner,
-    repo,
-    githubLabel
-  );
-
-  if (alreadyExists === null || alreadyExists === undefined) {
-    handler.createLabel(
-      github,
-      owner,
-      repo,
-      githubLabel,
-      colorsData[colorNumber].color
-    );
-  } else {
-    console.log(
-      'This label already exists on the repository, will check if it also exists on the issue'
-    );
-  }
 
-  if (labelsOnIssue) {
-    const found = labelsOnIssue.find(
-      (element: {name: string}) => element.name === githubLabel
-    );
-    const cleanUpOtherLabels = labelsOnIssue.filter(
-      element =>
-        element.name.startsWith('api') &&
-        element.name !== found?.name &&
-        element.name !== autoDetectedLabelAdded
-    );
-    if (found) {
-      console.log('This label already exists on this issue');
-      return;
+    if (labelsOnIssue) {
+      const found = labelsOnIssue.find(
+        (element: {name: string}) => element.name === githubLabel
+      );
+      const cleanUpOtherLabels = labelsOnIssue.filter(
+        element =>
+          element.name.startsWith('api') &&
+          element.name !== found?.name &&
+          element.name !== autoDetectedLabel
+      );
+      if (found) {
+        console.log('This label already exists on this issue');
+        return;
+      } else {
+        await handler.addLabels(github, owner, repo, issueNumber, [
+          githubLabel,
+        ]);
+        wasNotAdded = false;
+      }
+      for (const dirtyLabel of cleanUpOtherLabels) {
+        await github.issues
```

Comment:
I really like this style of calling the API where you don't rely on wrapper functions.